docs/ref/models/fields.txt

```diff
@@ -2266,6 +2211,119 @@ accepted by :class:`ForeignKey`, plus one extra argument:
 See :doc:`One-to-one relationships </topics/db/examples/one_to_one>` for usage
 examples of ``OneToOneField``.
 
+.. _lazy-relationships:
+
+Lazy relationships
+------------------
+
+Lazy relationships allow referencing models by their names (as strings) or
+creating recursive relationships. Strings can be used as the first argument in
+any relationship field to reference models lazily. A lazy reference can be
+either :ref:`recursive <recursive-relationships>`,
+:ref:`relative <relative-relationships>` or
+:ref:`absolute <absolute-relationships>`.
+
+.. _recursive-relationships:
+
+Recursive
+~~~~~~~~~
+
+To define a relationship where a model references itself, use ``"self"`` as the
+first argument of the relationship field::
+
+    from django.db import models
+
+
+    class Manufacturer(models.Model):
+        suppliers = models.ManyToManyField("self", symmetrical=False)
+        # ...
+
+When used in an :ref:`abstract model <abstract-base-classes>`, the recursive
+relationship will point to concrete model subclass.
+
+.. _relative-relationships:
+
+Relative
+~~~~~~~~
+
+When a relationship needs to be created with a model that has not been defined
+yet, it can be referenced by its name rather than the model object itself::
+
+    from django.db import models
+
+
+    class Car(models.Model):
+        manufacturer = models.ForeignKey(
+            "Manufacturer",
+            on_delete=models.CASCADE,
+        )
+        # ...
+
```

Comment:
Having introduced a `name` field in the earlier example, I think we should include it here as well, or else replace the `name` with `# ...`, or something like that, as done in other examples

Reply:
I'm in favour of removing the `name` field for consistency not only with the `Car` model, but also with `Manufacturer` model, which should also have a `name`, but has `# ...` to mean "other fields here"

Reply:
Sounds good, would you like to fetch the latest and push a new revision with all changes squashed?  Thank you!

Reply:
@cliff688 I have reviewed the latest revno available and I still see the `name = ...` definition. I understood your preference was to remove this field, did I misunderstand you? :thinking:

Reply:
Oh, I think when I was reviewing the changes here in patch form, the `Car` model on `FK` seemed close to the one in Lazy Relationships, so I thought we should keep them consistent, thinking they were in the same context. So, yes, you understood me (thank you), but I had misunderstood the placing of the two `Car` models, so that the issue I had imagined was there disappeared when viewing the file locally as a whole

Reply:
As for `Manufacturer.name` I think that's harmless.

Reply:
Thank you @cliff688, I pushed final tweaks, I will merge when CI is green again. Amazing work! :trophy: